notice.go

```diff
@@ -0,0 +1,82 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/elastic/go-licenser/licensing"
+)
+
+func doNotice(path string, params runParams) error {
+	// TODO: potentially use git to extract the date.
+	var startDate, _ = time.Parse("2006", params.noticeYear)
+	if noticeYear == "" {
+		startDate = time.Now()
+	}
+	if err := filepath.Walk(path,
+		func(p string, info os.FileInfo, err error) error {
```

Comment:
Should we move this anon function to a named function. The code looks a bit complex here :(

Reply:
I've added a comment